### PR TITLE
Update the TC responsibilities w.r.t. donations

### PR DESCRIPTION
**Pull request by bhs**
This change is as discussed at a GC meeting in December. Happy to take feedback and/or discuss the proposal if the TC is concerned about feasibility. cc @open-telemetry/technical-committee @open-telemetry/governance-committee

### CODE REVIEW

**Review by carlosalberto — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Review by alolita — APPROVED** [review]
@bhs +1 to outlining how large donations are handled by OpenTelemetry.
It may also be useful to add some detail about 
* how to contact the TC when submitting a large code donation 
* outlining requirements such as an agreement to change of copyright to reflect the OpenTelemetry Authors and code license being compatible with Apache 2.0.
* Code maintenance requirements

**Comment by bhs** [comment]
@alolita great suggestions. I updated this PR to include more guidance (per your comment) and moved some of that guidance into `CONTRIBUTING.md` where I'd look for it if I were an outside donator.

**Comment by bhs** [comment]
I can theoretically merge this given the approvals, but I changed the PR in meaningful ways since those approvals came through. Would love an explicit 👍 from an approver before this gets merged.

File: CONTRIBUTING.md
Comment (tigrannajaryan) [inline CONTRIBUTING.md:246]: Nit: I have never seen abbreviation "ASL 2" before. Not sure what the purpose of using it here is.

**Review by tigrannajaryan — APPROVED** [review]

File: CONTRIBUTING.md
Comment (bhs) [reply CONTRIBUTING.md:246]: interesting – I have seen it, but I am also glad to remove it :)  Will do.

**Pull request merged**
